### PR TITLE
feat: add SelectionGroup collection for Menu

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-types/shared/src/collections.d.ts [hunk 1]
```diff
@@ -49,6 +50,8 @@ export interface SectionProps<T> {
   items?: Iterable<T>
 }
 
+export interface SelectionGroupProps<T> extends Omit<CollectionBase<T>, 'disabledKeys'>, Omit<MultipleSelection, "disabledKeys"> {}
```
Comment (intergalacticspacehighway) [inline packages/@react-types/shared/src/collections.d.ts:53]: We can allow passing disabledKeys to SelectionGroup but then we'll have to add a way to set disabled keys state in parent's selection manager as it controls the keyboard navigation. Let me know If there's another way!

File: packages/@react-stately/collections/src/Section.ts [hunk 2]
```diff
@@ -41,12 +41,13 @@ Section.getCollectionNode = function* getCollectionNode<T>(props: SectionProps<T
       } else {
         let items: PartialNode<T>[] = [];
         React.Children.forEach(children, child => {
+          // @ts-ignore - Todo: find a better approach to conditionally check the type
```
Comment (intergalacticspacehighway) [inline packages/@react-stately/collections/src/Section.ts:44]: This is a bit hacky, need some suggestion how to make it better!

File: packages/@react-spectrum/menu/stories/MenuTrigger.stories.tsx [hunk 3]
```diff
@@ -527,6 +528,47 @@ storiesOf('MenuTrigger', module)
...
+            </Item>
+            <Item key="8">
+              Eight
+            </Item>
+            <Item key="9">
+              Nine
+            </Item>
+          </SelectionGroup>
+        </Section>
+        <Item key="4">
```
Comment (snowystinger) [inline packages/@react-spectrum/menu/stories/MenuTrigger.stories.tsx:561]: It's confusing that the section/selectiongroup has ended, not a fault of your PR, it's working pretty well, just noting something we'll need to solve for
I'll bring it up with design and see if we can't get some guidelines around mixing Sections with Items and what SelectionGroups can go around. We may decide to handle that in follow-up work.

**Review by snowystinger — COMMENTED** [review]
Thanks so much for creating this, enormously helpful

**Comment by snowystinger** [comment]
Just to be clear, i know you just submitted this so that we'd have your prior work to go off of, these are mostly discussion points and figured this would be an ok place to start those discussions

**Comment by snowystinger** [comment]
Closing because we'll probably handle this differently in RAC and we can still refer back here for that work

**Pull request closed without merging**